api/model/src/main/java/org/projectnessie/model/Util.java

```diff
@@ -44,15 +43,70 @@ private Util() {}
   public static final char REF_HASH_SEPARATOR = '@';
 
   /**
-   * Convert from path encoded string to normal string.
+   * Convert from path encoded string to normal string, supports all Nessie Spec versions.
+   *
+   * <p>The {@code encoded} parameter is split at dot ({@code .}) characters.
+   *
+   * <p>Legacy compatibility: a dot ({@code .}) character can be represented using ASCII 31 (0x1F)
+   * or ASCII 0 (0x00).
+   *
+   * <p>Escaping, for compatibility w/ <a
+   * href="https://jakarta.ee/specifications/servlet/6.0/jakarta-servlet-spec-6.0.html#uri-path-canonicalization">Jakarta
+   * Servlet Specification 6, URI Path Canonicalization</a>. This is available with Nessie Spec
+   * version 2.2.0, as advertised via {@link NessieConfiguration#getSpecVersion()}.
+   *
+   * <ul>
+   *   <li>Fact: two consecutive dots ({@code ..}) would represent an <em>empty</em> namespace
```

Comment:
Re: the specific encoding method... just my 2 cents: I think we can make it visually acceptable by using a rare char for escaping, e.g. `~`. So a literal period would be `~.`. Since `~` used to be a valid char, we can indicate the new encoding scheme by using `.` as the prefix, so examples:

`some.namespace` 
`some%1Dnamespace` - old escaping
`.some~.namespace` - new escaping
`some~namespace` - old literal `~`
`.some~~namespace` - new literal `~`

I hope in this case most textual respresentations are human-readable.

This is just for discussion. WDYT?

Reply:
`~.` is ambiguous - `~` could be the last char of the element.

Reply:
Leading `.` means all `~` are special.

Reply:
The leading dot to indicate the new encoding scheme is an interesting idea imho.

2 suggestions:

1. Can we use an escape char that doesn't get url-encoded? `~` becomes `%5E`. These chars remain the same: `-`, `*`, and `_` 
2. Could we use the leading dot only if there are escaped sequences? This way if there is no leading dot we know that it's either old-style encoding, or a simple namespace that doesn't need escaping.

Reply:
1) absolutely... choosing a char may be tricky, `-` and `_` are common in names and `*` is special in shell :)

2) yes, that's the idea. It would make hand-writing keys easy (I hope).... e.g. in CLI args. 

Minor point: `.some.name` and `some.name` would be equivalent (`.` is the separator here). We can choose which one is the "canonical" form.

Reply:
Described the new approach in the PR description - hope I got all the dots right.

Reply:
also `toCanonicalString` as a "weak" variant of the escaped one